src/utils.ts

```diff
@@ -63,14 +63,37 @@ export async function signPayload(peerId: PeerId, payload: bytes): Promise<bytes
   return peerId.privKey.sign(payload);
 }
 
+export async function getPeerIdFromPayload(payload: bytes): Promise<PeerId> {
+  const decodedPayload = await decodePayload(payload);
+  return await PeerId.createFromPubKey(Buffer.from(decodedPayload.identityKey));
+}
+
+async function decodePayload(payload: bytes){
+  const NoiseHandshakePayload = await loadPayloadProto();
+  return NoiseHandshakePayload.toObject(
+    NoiseHandshakePayload.decode(payload)
+  );
+}
+
 export const getHandshakePayload = (publicKey: bytes ) => Buffer.concat([Buffer.from("noise-libp2p-static-key:"), publicKey]);
 
 async function isValidPeerId(peerId: bytes, publicKeyProtobuf: bytes) {
   const generatedPeerId = await PeerId.createFromPubKey(publicKeyProtobuf);
   return generatedPeerId.id.equals(peerId);
 }
 
-export async function verifySignedPayload(noiseStaticKey: bytes, plaintext: bytes, peerId: bytes) {
+/**
+ * Verifies signed payload and returns peer id that has sent the payload.
+ * @param {bytes} noiseStaticKey - owner's noise static key
+ * @param {bytes} plaintext - encoded payload
+ * @param {PeerId} remotePeer - (optional) owner's libp2p peer ID
+ * @returns {Promise<PeerId>} - peer ID of payload owner
+ */
+export async function verifySignedPayload(
+  noiseStaticKey: bytes,
+  plaintext: bytes,
+  remotePeer?: PeerId
+): Promise<PeerId> {
```

Comment:
It doesn't make sense for this method to return PeerId, it's called verifySignedPayload it can only return boolean or void.

It makes sense for this method to receive decodedPayload instead of plaintext